docs/additional-examples/evm-swap.md

```diff
@@ -0,0 +1,656 @@
+      authSig: JSON.parse(authSig),
+      chain: chainBCondition.chain,
+    });
+
+    console.log("chainAConditionsPass: ", chainAConditionsPass, "chainBConditionsPass: ", chainBConditionsPass);
+```
+
+Now our action checks if the conditions are passed on both chains, it takes `authSig` as a parameter here and then logs the result of both condition checks. We'll look into auth sigs while we learn to execute lit actions.
+
+For generating signatures inside our Lit Action, we'll use `Lit.Actions.signEcdsa`. This method allows us to use a pkp (which lets an action generate a sign) for generating signatures. Our Lit Action will now decide based on checked conditions and generate appropriately signed transactions which we can send to the blockchain later.
+
+```js
+    if (chainAConditionsPass && chainBConditionsPass) {
+      await generateSwapTransactions();
+      return;
+    }
+
+    if (chainAConditionsPass) {
+      await Lit.Actions.signEcdsa({
+        toSign: hashTransaction(chainAClawbackTransaction),
+        publicKey: pkpPublicKey,
+        sigName: "chainASignature",
+      });
+      Lit.Actions.setResponse({
+        response: JSON.stringify({
+          chainATransaction: chainAClawbackTransaction,
+        }),
+      });
+      return;
+    }
+  
+    if (chainBConditionsPass) {
+      await Lit.Actions.signEcdsa({
+        toSign: hashTransaction(chainBClawbackTransaction),
+        publicKey: pkpPublicKey,
+        sigName: "chainBSignature",
+      });
+      Lit.Actions.setResponse({
+        response: JSON.stringify({
+          chainBTransaction: chainBClawbackTransaction,
+        }),
+      });
+      return;
+    }
+
+    const generateSwapTransactions = async () => {
+      await LitActions.signEcdsa({
+        toSign: hashTransaction(chainATransaction),
+        publicKey: pkpPublicKey,
+        sigName: "chainASignature",
+      });
+      await LitActions.signEcdsa({
+        toSign: hashTransaction(chainBTransaction),
+        publicKey: pkpPublicKey,
+        sigName: "chainBSignature",
+      });
+      Lit.Actions.setResponse({
+        response: JSON.stringify({ chainATransaction, chainBTransaction }),
+      });
+    };
+```
+
+As everything on the blockchain works on hexadecimal hashes, we need to hash our transactions as well.
+
+```js
+    const hashTransaction = (tx) => {
+      return ethers.utils.arrayify(
+        ethers.utils.keccak256(
+          ethers.utils.arrayify(ethers.utils.serializeTransaction(tx)),
+        ),
+      );
+    }
+
+    Lit.Actions.setResponse({ response: "Conditions for swap not met!" });
+  }
+go();
+```
+
+A must `Lit.Actions.setResponse` needs to be set before we are closing the execution of a Lit Action. This is the response which is shown after the execution of a Lit Action.
+
+Complete Lit Action can be found at [swapAction.js](https://github.com/anshss/lit-evm-swap/blob/main/lit/swapAction.js).
+
+## 3) Implementing Mint/Grant/Burn with SDK
```

Comment:
This goes back to my earlier question, but m/g/b is broken right now right?

Also, is the point of this section to go over the full flow start to finish?

Reply:
Our SDK does not support mint/grant/burn calls at this moment so we can keep this alternate approach for a time even after the fix.

The only change we'll later need to make would be in this [function](https://github.com/LIT-Protocol/docs/pull/330/commits/e1a8353388296a6bbdbda28fb4f81d9681c27d33#diff-0e36021eb2e9b13aa34e50ec77474d75d50bab8d18cba35c954841b81d9efee7R342).